src/Commands/GenerateModule.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+namespace FOP\Console\Commands;
+
+use FOP\Console\Command;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+
+/**
+ * This command is a working exemple.
+ */
+final Class GenerateModule extends Command
+{
+
+    protected static $tabHook=array();
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:console:generate-module')
+            ->setDescription('create a new module')
+            ->setHelp('This command instantiate every files and directores of Symfony/Prestashop in Console Context')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $filesystem = new Filesystem();
+
+        $io->text(sprintf("\033\143"));
+
+        $io->title('This program was carried out by FriendsOfPresta and OHweb');
+        $io->note("It work better on unix system");
+        $io->warning('Be careful if you run this program it may erase data, be sure to specify an unused module name or you know what you are doing');
+        $confirm = $io->ask('Would like to continue ? (yes/no)', 'no');
+        while ($confirm != "yes" && $confirm != "no"){
+            $confirm = $io->ask('Please answer with "yes" or no "no"', 'no');
+        }
+        if ($confirm == "no"){
+            $io->text("End of the program ...");
+            return;
+        }
+
+        $io->text(sprintf("\033\143"));
+        $io->section('Edition of the entry point');
+
+        $name = $io->ask('Give the name of your module','oh_exemplemodule');
+        $name = str_replace(' ', '_', $name);
+        $fileName = strtolower($name);
+        $filenameFirstletterCaps = ucfirst($name);
+        $filenameCamel = ucwords(strtolower($name));
+        $description = $io->ask("Give the description of the module");
+        $author = $io->ask("Give the author of the module",'OHweb');
+
+        $io->text(sprintf("\033\143"));
+        $io->section('File generation');
+
+        try {

```

Comment:
Is this necessary to create all the file and directories before asking the user what he want to generate ?

Reply:
Yes I need at least the filename provided by the user.